packages/amplify-codegen/src/utils/validateDartSDK.js

```diff
@@ -0,0 +1,26 @@
+const yaml = require('js-yaml');
+const path = require('path');
+const fs = require('fs-extra');
+const semver = require('semver');
+
+const pubspecFile = 'pubspec.yaml';
+
+function validateDartSDK(context, projectRoot) {
+  try {
+    const config = yaml.load(fs.readFileSync(path.join(projectRoot, pubspecFile), 'utf8'));
+    const version = semver.minVersion(config.environment.sdk);
```

Comment:
Does this semver package handle [caret syntax](https://dart.dev/tools/pub/dependencies#caret-syntax)?

Reply:
They do handle the caret version. https://www.npmjs.com/package/semver#ranges